doc/deprecated.rst

```diff
@@ -39,6 +39,9 @@ Tags
 * The ``filter`` tag is deprecated in Twig 2.9. Use the ``apply`` tag instead
   (the ``Twig\TokenParser\FilterTokenParser`` classes is also deprecated).
 
+* Adding an ``if`` condition on a ``for`` tag is deprecated in Twig 2.10. Use a
+``filter`` filter instead.
```

Comment:
I would say ``use a `filter` filter or a condition inside the `if` instead`` (if your condition depends on a variable updated inside the loop, you cannot migrate to a filter)

Reply:
same in the deprecation warning

Reply:
Done